### ISSUE TITLE
Add option to set and Update Combobox Available options from script

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#476]
<sup>Issue created by JBKacerovsky as [Bitbucket Issue #&#8203;337](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/337) on 2020.08.26 22:49.</sup>
I am trying to use the Combobox UI extension to allow for the selection of input files in a DataViewer app I am building. To do the I want to set AvailableOptions from a script \(e.g. file names in a folder\). I can set AvailableOptions from my script \(I can see it update in the inspector\), however the in game dropdown list does not change.

I was able to work around this issue by including the following short void in the Combobox script. I think something along those lines could be a useful option to add. 

```c
        public void SetAvailableOptions(List<string> newOptions)
        {
            AvailableOptions = newOptions;
            RebuildPanel();
        }
```

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.08.27 18:24](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/337#comment-58612376), @SimonDarksideJ commented:</sup>
[https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/pull-requests/90](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/pull-requests/90)

Created which adds the following additional commands to manage the DropDown controls programmatically:

‌

```
> * AddItem - Adds a single item to the list
> * RemoveItem - Removes a single item from the list
> * SetAvailableOptions - Clears the current options and replaces with new list (array or List)
> * ResetItems - Clears all current options
```

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.08.27 18:26](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/337#comment-58612393), @SimonDarksideJ commented:</sup>
Docs also updated to reflect the new option and advises against modification of the \*AvailableOptions\* parameter.

Would have preferred to make the AvailableOptions parameter readonly, however, this would be a breaking change.

If you’re happy with the changes, I’ll get that merged in @{5f46d67dfdc3f5003ffac869}

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.08.27 19:06](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/337#comment-58612645), JBKacerovsky commented:</sup>
Yes, the changes look great!

Thanks for adding that in so quickly. The added options are really useful for the project I am working on!
